src/magicbox/MagicBox.ts

```diff
@@ -145,6 +145,13 @@ export class MagicBoxInstance {
       if (key === KEYBOARD.UP_ARROW || key === KEYBOARD.DOWN_ARROW) {
         return false;
       }
+      if (
+        this.suggestionsManager.hasFocus &&
+        this.suggestionsManager.hasPreviews &&
+        (key === KEYBOARD.LEFT_ARROW || key === KEYBOARD.RIGHT_ARROW)
+      ) {
+        return false;
+      }
```

Comment:
This prevents the keyboard from moving the cursor in the input box while suggestions or previews are focused. However, this doesn't prevent movement when there are no previews; keeping the same default behaviour.

Reply:
Might be good to extract that comment you just put there and "name" a function that describe what you just explained.


Something like: 

```
if(cursorShouldBeBlockedInInputWhileInPreviewMode()) {return false;}`
```

Might not be the perfect name, but I guess you get the point I'm trying to make ? 

Because comments like this on PR will be forgotten very quickly ;)

Reply:
Good point.

`MagicBox`'s handling of the keyboard with `SuggestionsManager` is a bit confusing, so I'll introduce `shouldMoveInSuggestions(key)` and simplify it.